### PR TITLE
feat: implement path validations & filters

### LINKED ISSUE #3930: Improve RequestValidation
See https://github.com/bisq-network/bisq2/pull/3929#discussion_r2422720324

@wodoro Would you be available to tackle that?

### PR DESCRIPTION
**Pull request by wodoro**
resolves #3930

## Summary by CodeRabbit

- **New Features**
  - Request filtering for REST and WebSocket endpoints using regex-based allow/deny lists; invalid requests are rejected with clear HTTP error responses.

- **Documentation**
  - Configuration comments clarified: whitelist/blacklist entries are regex patterns and examples updated.

- **Refactor**
  - Centralized and unified request validation across services; removed the old standalone validator utility.

- **Tests**
  - Added comprehensive unit tests for path validation, whitelist/blacklist rules, and edge cases.

### CODE REVIEW

File: http-api/src/main/java/bisq/http_api/HttpApiService.java [hunk 1]
```diff
@@ -124,7 +127,8 @@ public HttpApiService(RestApiService.Config restApiConfig,
                         explorerRestApi,
                         paymentAccountsRestApi,
                         reputationRestApi,
-                        userProfileRestApi);
+                        userProfileRestApi,
+                        GlobalRequestValidatorFilter.from(webSocketConfig));
```
[the remark this answers is not in the document]
Comment (wodoro) [reply http-api/src/main/java/bisq/http_api/HttpApiService.java]: I dont understand this

File: http-api/src/main/java/bisq/http_api/HttpApiService.java [hunk 2]
```diff
@@ -136,7 +140,8 @@ public HttpApiService(RestApiService.Config restApiConfig,
                         tradeService,
                         userService,
                         bisqEasyService,
-                        openTradeItemsService));
+                        openTradeItemsService,
+                        WebSocketRequestValidator.from(webSocketConfig)));
```
[the remark this answers is not in the document]
Comment (wodoro) [reply http-api/src/main/java/bisq/http_api/HttpApiService.java:144]: I dont understand this either

File: http-api/src/main/java/bisq/http_api/validator/RequestValidator.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    private final Set<Pattern> endpointBlacklistPatterns;
+
+    public RequestValidator(List<String> whitelistPatterns, List<String> blacklistPatterns) {
+        endpointWhitelistPatterns = whitelistPatterns.stream()
+                .map(Pattern::compile)
+                .collect(java.util.stream.Collectors.toSet());
+        endpointBlacklistPatterns = blacklistPatterns.stream()
+                .map(Pattern::compile)
+                .collect(java.util.stream.Collectors.toSet());
+    }
```
[the remark this answers is not in the document]
Comment (wodoro) [reply http-api/src/main/java/bisq/http_api/validator/RequestValidator.java:25]: I think this one is ok ?

File: http-api/src/main/java/bisq/http_api/validator/WebSocketRequestValidator.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+public class WebSocketRequestValidator {
+    private static final Set<String> METHOD_WHITELIST = Set.of("GET", "POST", "DELETE", "PUT", "PATCH");
+    private final RequestValidator requestValidator;
+
+    public WebSocketRequestValidator(List<String> whitelistPatterns, List<String> blacklistPatterns) {
+        this.requestValidator = new RequestValidator(whitelistPatterns, blacklistPatterns);
+    }
+
+    public String validateRequest(WebSocketRestApiRequest request) {
```
Comment (HenrikJannsen) [inline http-api/src/main/java/bisq/http_api/validator/WebSocketRequestValidator.java:19]: Add `@Nullable` annotation

File: http-api/src/main/java/bisq/http_api/validator/RequestValidator.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+    public boolean isValidPath(String path) {
+        URI uri;
+        try {
+            // path may come from websocket where it's not parsed yet
+            // this also decodes URI components
+            uri = new URI(path);
```
Comment (HenrikJannsen) [inline http-api/src/main/java/bisq/http_api/validator/RequestValidator.java]: I guess we should add a `normalize()` call here.

File: http-api/src/main/java/bisq/http_api/validator/RequestValidator.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        } catch (Exception e) {
+            return false;
```
Comment (HenrikJannsen) [inline http-api/src/main/java/bisq/http_api/validator/RequestValidator.java:32]: Better add logs in all failure cases (also below when returning false)

File: http-api/src/main/java/bisq/http_api/validator/RequestValidator.java [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        }
+        String normalizedPath = uri.getPath();
+        if (normalizedPath == null) {
```
Comment (HenrikJannsen) [inline http-api/src/main/java/bisq/http_api/validator/RequestValidator.java]: Maybe add length check for normalizedPath?

File: http-api/src/main/java/bisq/http_api/validator/RequestValidator.java [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+import java.net.URI;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+class RequestValidator {
+    // Only allow alphanumeric characters and some special characters like /, -, _
+    // Does not allow dots to avoid path traversal attacks (example: '../')
+    private static final Pattern allowedCharacters = Pattern.compile("^[a-zA-Z0-9/_\\-?=,&]*$");
```
Comment (HenrikJannsen) [inline http-api/src/main/java/bisq/http_api/validator/RequestValidator.java]: Use Uppercase for static final. Maybe SAFE_PATH?

File: http-api/src/main/java/bisq/http_api/validator/GlobalRequestValidatorFilter.java [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+public class GlobalRequestValidatorFilter implements ContainerRequestFilter {
+    private final RequestValidator validator;
+
+    public GlobalRequestValidatorFilter(List<String> whitelist, List<String> blacklist) {
+        this.validator = new RequestValidator(whitelist, blacklist);
+    }
+
+    @Override
+    public void filter(ContainerRequestContext requestContext) throws IOException {
+        String path = requestContext.getUriInfo().getRequestUri().getPath();
```
Comment (HenrikJannsen) [inline http-api/src/main/java/bisq/http_api/validator/GlobalRequestValidatorFilter.java]: Why not use the URI and pass that?

`URI requestUri = requestContext.getUriInfo().getRequestUri();`
And later we don't need to construct the URI. The normalize should still be added.

File: http-api/src/main/java/bisq/http_api/validator/GlobalRequestValidatorFilter.java [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+import jakarta.ws.rs.container.ContainerRequestContext;
+import jakarta.ws.rs.container.ContainerRequestFilter;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.ext.Provider;
+
+import java.io.IOException;
+import java.util.List;
+
+@Provider
+public class GlobalRequestValidatorFilter implements ContainerRequestFilter {
```
Comment (HenrikJannsen) [inline http-api/src/main/java/bisq/http_api/validator/GlobalRequestValidatorFilter.java]: We could add `@Priority(Priorities.AUTHENTICATION)` so that it will run before the other registered entries as AUTHENTICATION is highest prio.

Maybe avoid the `Global` in the name. Maybe `HttpApiRequestFilter` as HttpApi is our containing name space.
